Unable to build SDK inside of an ARM64 WSL/Ubuntu 20.04 Environment

**Environment:**

```
gcc   version: gcc (Ubuntu/Linaro 7.5.0-6ubuntu2) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
cmake -DMAC_ARCH=x86_64 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PACKAGE_TYPE=deb -DCMAKE_CXX_FLAGS= DBUILD_SHARED_LIBS=OFF ..
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CMAKE_SYSTEM_INFO_FILE: Platform/Linux
-- CMAKE_SYSTEM_NAME:      Linux
-- CMAKE_SYSTEM_PROCESSOR: aarch64
-- CMAKE_SYSTEM:           Linux-5.15.79.1-microsoft-standard-WSL2
-- CMAKE_SYSTEM_VERSION:   5.15.79.1-microsoft-standard-WSL2
-- CMAKE_BUILD_TYPE:       Debug
-- TARGET_ARCH:            aarch64
-- CMAKE_CXX_COMPILER_ID:  GNU
-- /etc/os_release : NAME=Ubuntu
-- /etc/os_release : VERSION=20.04.3 LTS (Focal Fossa)
-- /etc/os_release : ID=ubuntu
-- /etc/os_release : ID_LIKE=debian
-- /etc/os_release : PRETTY_NAME=Ubuntu 20.04.3 LTS
-- /etc/os_release : VERSION_ID=20.04
-- /etc/os_release : HOME_URL=https://www.ubuntu.com/
-- /etc/os_release : SUPPORT_URL=https://help.ubuntu.com/
-- /etc/os_release : BUG_REPORT_URL=https://bugs.launchpad.net/ubuntu/
-- /etc/os_release : PRIVACY_POLICY_URL=https://www.ubuntu.com/legal/terms-and-policies/privacy-policy
-- /etc/os_release : VERSION_CODENAME=focal
-- /etc/os_release : UBUNTU_CODENAME=focal
Building Debug ...
-- Not Found Tcmalloc
-- -- PAL implementation: CPP11
-- -- SDK version: EVT-3.4.348.0
-- Found CURL: /usr/lib/aarch64-linux-gnu/libcurl.so (found version "7.68.0")
Adding gtest
Adding gmock
-- Building with Azure Monitor support
-- Looking for pthread.h
```

**Repro steps**
1 - ```git clone --single-branch --branch v3.5.321  --recurse-submodules https://github.com/microsoft/cpp_client_telemetry $1dsExtPath```
2 - ```wsl.exe -d Ubuntu-20.04 --user root -- bash -c "$1dsWslPath/build.sh"```

**What is the expected behavior?**
SDK builds and I am able to use it in my project.

**What is the actual behavior?**
Build breaks in VariantTypes.hpp:

```
/mnt/c/dev/myprojectname/external/cpp_client_telemetry/lib/include/public/VariantType.hpp: In static member function ‘static std::__cxx11::string Microsoft::Applications::Events::Variant::escape(const string&)’:
/mnt/c/dev/myprojectname/external/cpp_client_telemetry/lib/include/public/VariantType.hpp:399:28: error: comparison is always true due to limited range of data type [-Werror=type-limits]
                 if ('\x00' <= *c && *c <= '\x1f') {
                     ~~~~~~~^~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [lib/CMakeFiles/mat.dir/build.make:63: lib/CMakeFiles/mat.dir/decorators/BaseDecorator.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:175: lib/CMakeFiles/mat.dir/all] Error 2
```

**Additional context.**
Unsure if I'm holding something wrong here or in an unsupported state. Tried going the vcpkg route originally but that turned out to be not fully supported on ARM Linux.

--- a/lib/include/public/VariantType.hpp
+++ b/lib/include/public/VariantType.hpp
@@ -396,7 +396,7 @@ public:
             case '\r': o << "\\r"; break;
             case '\t': o << "\\t"; break;
             default:
-                if ('\x00' <= *c && *c <= '\x1f') {
+                if ('\x00' <= static_cast<signed char>(*c) && *c <= '\x1f') {
                     o << "\\u"
                         << std::hex << std::setw(4) << std::setfill('0') << (int)*c;
                 }

